From 3346526306298ec9de4ff493533ce372e7267b1a Mon Sep 17 00:00:00 2001
From: Nicolai Stange <nstange@suse.de>
Date: Tue, 14 Jul 2026 06:23:17 +0200
References: jsc#PED-15986,bsc#1262160
Patch-mainline: Never, downstream workaround to meet certification requirements.
Subject: [PATCH] crypto/xxhash: set downstream CRYPTO_ALG_FIPS_UNAPPROVED

It's been requested to somehow mark xxhash64 as unapproved for FIPS.
xxhash64 must remain usable in FIPS mode, so its associated testmgr
entry's ->fips cannot get flipped to false. Set the new
CRYPTO_ALG_FIPS_UNAPPROVED in the ->cra_flags instead.

Signed-off-by: Nicolai Stange <nstange@suse.de>
---
 crypto/xxhash_generic.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/crypto/xxhash_generic.c
+++ b/crypto/xxhash_generic.c
@@ -79,7 +79,8 @@ static struct shash_alg alg = {
 		.cra_name	 = "xxhash64",
 		.cra_driver_name = "xxhash64-generic",
 		.cra_priority	 = 100,
-		.cra_flags	 = CRYPTO_ALG_OPTIONAL_KEY,
+		.cra_flags	 = (CRYPTO_ALG_OPTIONAL_KEY |
+				    CRYPTO_ALG_FIPS_UNAPPROVED),
 		.cra_blocksize	 = XXHASH64_BLOCK_SIZE,
 		.cra_ctxsize	 = sizeof(struct xxhash64_tfm_ctx),
 		.cra_module	 = THIS_MODULE,
